api/src/main/java/org/eclipse/microprofile/config/spi/ConfigValue.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright (c) 2020 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package org.eclipse.microprofile.config.spi;
+
+import java.util.Objects;
+
+/**
+ * The ConfigValue is a metadata object that holds additional information after the lookup of a configuration.
+ * <p>
+ *
+ * Right now, it is able to hold information like the configuration name, value, the Config Source from where
+ * the configuration was loaded, the ordinal of the Config Source and a line number from where the configuration was
+ * read if exists.
```

Comment:
What do you need these kind of metadata for?
The only use case we had for this information so far is a summary page which kind of enlists all the ConfigSources and values therein for diagnostic reasons. But this can be perfectly done even with the mp-config-1.0 API already. So what is the additional value of completely changing the whole design (in likely backward incompatible ways)

Reply:
Well, we have to start somewhere. Yes, the initial use case would be to provide debug information about the key lookup. Idea was to expand this in the future to add more things.